### DISCUSSION TITLE
Developing a custom Execution Module

### DISCUSSION THREAD

**Discussion by hmita** [bcnmy/biconomy-client-sdk#516] [Q&A] [answered]
Hi, for our project we would like to add a custom Execution Module to the Smart Account.

We want to do a few things, but mainly we want to be able to receive payment from users' smart account like a BNPL model.
Basically what we are thinking of right now is to have users' funds to be "allowed" for use, and for us to be able to receive the payment later on whenever it is appropriate.

Either way, we were looking for some examples or any documentation on how to implement a custom execution module, but was told there aren't any.
Any help would be appreciated, and we are not restricted to just custom execution modules (it just seemed to be what we are looking for).

**Comment by livingrockrises** [comment, marked as answer]
i. I think you could issue a session that is allowed to transfer erc20 / eth and make use of it for PL

ii. execution module -> once gets plugged in it has to call execTransactionFromModule on the smart account. you can implement some access control in this module limited to allowed currency transfer as module can call anything on SA otherwise. 

flow could be like 
your admin eoa -> some exec method on executor -> execTxFromModule on SA -> erc20/eth transfer
